TimescaleDB 2.7 Release Notes (#1091)
timescale/docs: timescaledb/overview/release-notes/index.md

follow these [setup instructions][distributed-hypertables-setup].

### What's new in TimescaleDB 2.7:

You can recreate a continuous aggregate to benefit from the improvement, if the
original data already exists.

- Improved continuous aggregates performances and lifted several limitations (by removing reaggregation)

This new continuous aggregate format is faster than the previous version and also
because continuous aggregates no longer store partials, and re-aggregation is removed, 
it is now possible to create indexes on continuous aggregates columns.
This can further improve performance.
There are no longer any restrictions on which types of aggregates you can use
with continuous aggregates. All of these are now available:

* aggregates with DISTINCT
* aggregates with FILTER
* aggregates with FILTER in HAVING clause
* aggregates without combine function
* ordered-set aggregates
* hypothetical-set aggregates

The existing continuous aggregates are safe to use and continue to work as before. 
They are still supported, however they do not benefit from the performance advantages 
of the new format.

By default, TimescaleDB 2.7 and later uses the new continuous aggregate format.

> We recommend that you recreate you existing continuous aggregates to take advantage 

Reviewer: ```suggestion
We recommend that you recreate your existing continuous aggregates to take advantage 
```

follow these [setup instructions][distributed-hypertables-setup].

### What's new in TimescaleDB 2.7:

You can recreate a continuous aggregate to benefit from the improvement, if the
original data already exists.

- Improved continuous aggregates performances and lifted several limitations (by removing reaggregation)

This new continuous aggregate format is faster than the previous version and also
because continuous aggregates no longer store partials, and re-aggregation is removed, 
it is now possible to create indexes on continuous aggregates columns.
This can further improve performance.
There are no longer any restrictions on which types of aggregates you can use
with continuous aggregates. All of these are now available:

* aggregates with DISTINCT
* aggregates with FILTER
* aggregates with FILTER in HAVING clause
* aggregates without combine function
* ordered-set aggregates
* hypothetical-set aggregates

The existing continuous aggregates are safe to use and continue to work as before. 
They are still supported, however they do not benefit from the performance advantages 
of the new format.

By default, TimescaleDB 2.7 and later uses the new continuous aggregate format.

> You can still create continuous aggregates using the previous format by setting 
> the `timescaledb.finalized` storage parameter to `FALSE` when you create the 
> continuous aggregate.

Author: ```suggestion
```

on [community Slack](https://slack.timescale.com/) for installation details and
follow these [setup instructions][distributed-hypertables-setup].

### What's new in TimescaleDB 2.7:

TODO: Product notes
- Continuous aggregates without reaggregation

Because we're not storing partials anymore and removed the
re-aggregation, now is be possible to create indexes on aggregated
columns in the materialization hypertable in order to improve the
performance even more.

Also removed restrictions on types of aggregates users can perform
with Continuous Aggregates:

* aggregates with DISTINCT
* aggregates with FILTER
* aggregates with FILTER in HAVING clause
* aggregates without combine function
* ordered-set aggregates
* hypothetical-set aggregates

> By default the new Continuous Aggregates will be created using this new
> format, but the previous version (with partials) will be supported.

Reviewer 2: ```suggestion
By default the new continuous aggregates are created using this new
format, but the previous version (with partials) is still supported.
```

on [community Slack](https://slack.timescale.com/) for installation details and
follow these [setup instructions][distributed-hypertables-setup].

### What's new in TimescaleDB 2.7:

> You can recreate a cagg to benefit from the improvement with the caveat that the original data need to exist  

Reviewer 2: ```suggestion
You can recreate a continuous aggregate to benefit from the improvement, if the
original data already exists.
```

> You can recreate a cagg to benefit from the improvement with the caveat that the original data need to exist  

Reviewer 3: Maybe we should say "still exists"? Because I think the biggest concern is trying to recreate a cagg over data dropped in a retention policy (whereas if the data _never_ existed, well you have no cagg anyway 😛 )

> There will be a migration path

Reviewer 2: This looks like a fragment?

- Continuous aggregates without reaggregation

> Since Continuous aggregates are not storing partials anymore and with also removing re-aggregation, 

Reviewer 4: I would skip the explanation or put it at the end to focus on the new capability that is offered.

> now it is possible to create indexes on aggregated columns in the materialization hypertable 

Reviewer 4: "columns in the continuous aggregate" would be clearer than "aggregated columns in the materialization hypertable" if that's the same thing.

> in order to improve the performance even more.

Reviewer 4: we should mention that the Continuous Aggregates are faster than in previous versions (even before creating an index)

> Since Continuous aggregates are not storing partials anymore and with also removing re-aggregation, 
> now it is possible to create indexes on aggregated columns in the materialization hypertable 
> in order to improve the performance even more.

Reviewer 2: ```suggestion
Because continuous aggregates no longer store partials, and re-aggregation is removed, 
it is now possible to create indexes on aggregated columns in the materialization hypertable.
This can further improve performance.
```

> We have also removed restrictions on types of aggregates users can perform
> with Continuous Aggregates:

Reviewer 2: ```suggestion
There are no longer any restrictions on which types of aggregates you can use
with continuous aggregates. All of these are now available:
```

* aggregates with DISTINCT
* aggregates with FILTER
* aggregates with FILTER in HAVING clause
* aggregates without combine function
* ordered-set aggregates
* hypothetical-set aggregates

> The existing Continuous Aggregates are safe and will continue to work as before. 
> They are still supported however they will not take the performance advantages of the new format.

Reviewer 2: ```suggestion
The existing continuous aggregates are safe to use and continue to work as before. 
They are still supported, however they do not benefit from the performance advantages 
of the new format.
```

> By default the continuous aggregate format from TimescaleDB 2.7 onward is the new one.
> Users can still create in the previous format by setting to false the storage paramater 
> named `timescaledb.finalized` during the creation of the Continuous Aggregate.

Reviewer 2: ```suggestion
By default, TimescaleDB 2.7 and later uses the new continuous aggregate format.
You can still create continuous aggregates using the previous format by setting 
the `timescaledb.finalized` storage parameter to `FALSE` when you create the 
continuous aggregate.
```

> We recommend users that want to take advantage of these new improvements to recreate their Continuous Aggregates.
> You will need the original data to be able to achieve this.
> If you don't have the original data, we will provide in a later TimescaleDB version a migration path 
> from the previous Continuous Aggregate format to the new one.

Reviewer 2: ```suggestion
We recommend that you recreate you existing continuous aggregates to take advantage 
of these improvements. To do this, you need the original data. A migration path for 
migrating to the new continuous aggregate format without the original data is intended 
in a future version of TimescaleDB.
```

on [community Slack](https://slack.timescale.com/) for installation details and
follow these [setup instructions][distributed-hypertables-setup].

### What's new in TimescaleDB 2.7:

You can recreate a continuous aggregate to benefit from the improvement, if the
original data already exists.

> - Continuous aggregates without reaggregation

Author: ```suggestion
- Improved continuous aggregates performances and lifted several limitations (by removing reaggregation)
```

> Because continuous aggregates no longer store partials, and re-aggregation is removed, 

Author: ```suggestion
This new continuous aggregate format is faster than the previous version and also
because continuous aggregates no longer store partials, and re-aggregation is removed, 
```

> it is now possible to create indexes on aggregated columns in the materialization hypertable.
> This can further improve performance.

Author: ```suggestion
it is now possible to create indexes on continuous aggregates columns.
This can further improve performance.
```

with continuous aggregates. All of these are now available:

* aggregates with DISTINCT
* aggregates with FILTER
* aggregates with FILTER in HAVING clause
* aggregates without combine function
* ordered-set aggregates
* hypothetical-set aggregates

The existing continuous aggregates are safe to use and continue to work as before. 
They are still supported, however they do not benefit from the performance advantages 
of the new format.

By default, TimescaleDB 2.7 and later uses the new continuous aggregate format.
You can still create continuous aggregates using the previous format by setting 
the `timescaledb.finalized` storage parameter to `FALSE` when you create the 
continuous aggregate.

We recommend that you recreate you existing continuous aggregates to take advantage 
of these improvements. To do this, you need the original data. A migration path for 
migrating to the new continuous aggregate format without the original data is intended 
in a future version of TimescaleDB.

<!-- <highlight type="note"> This release is low priority for upgrade. We recommend that you upgrade when you can. </highlight> -->

> <!-- <highlight type="important"> This release is medium priority for upgrade. We recommend that you upgrade at the next available opportunity. </highlight> -->

Author: ```suggestion
<highlight type="important"> This release is medium priority for upgrade. We recommend that you upgrade at the next available opportunity. </highlight>
```

original data already exists.

- Improved continuous aggregates performances and lifted several limitations (by removing reaggregation)

This new continuous aggregate format is faster than the previous version and also
because continuous aggregates no longer store partials, and re-aggregation is removed, 
it is now possible to create indexes on continuous aggregates columns.
This can further improve performance.
There are no longer any restrictions on which types of aggregates you can use
with continuous aggregates. All of these are now available:

* aggregates with DISTINCT
* aggregates with FILTER
* aggregates with FILTER in HAVING clause
* aggregates without combine function
* ordered-set aggregates
* hypothetical-set aggregates

The existing continuous aggregates are safe to use and continue to work as before. 
They are still supported, however they do not benefit from the performance advantages 
of the new format.

By default, TimescaleDB 2.7 and later uses the new continuous aggregate format.
You can still create continuous aggregates using the previous format by setting 
the `timescaledb.finalized` storage parameter to `FALSE` when you create the 

> continuous aggregate.

Author: @jfjoly what do you think about removing this and documenting it for internal purposes only?

> continuous aggregate.

Reviewer 4: Ok for leaving this out from the release notes, thanks!

We recommend that you recreate you existing continuous aggregates to take advantage 
of these improvements. To do this, you need the original data. A migration path for 
migrating to the new continuous aggregate format without the original data is intended 
in a future version of TimescaleDB.

> TODO: Priority selection

Author: ```suggestion
```

</highlight>

## Release notes

> TODO:
> ## 2.7 (2022-05-XX)

Author: ```suggestion
## 2.7.0 (2022-05-24)

This release adds major new features since the 2.6.1 release.
We deem it moderate priority for upgrading.

This release includes these noteworthy features:

* Optimize continuous aggregate query performance and storage
* The following query clauses and functions can now be used in a continuous
  aggregate: FILTER, DISTINCT, ORDER BY as well as [Ordered-Set Aggregate](https://www.postgresql.org/docs/current/functions-aggregate.html#FUNCTIONS-ORDEREDSET-TABLE)
  and [Hypothetical-Set Aggregate](https://www.postgresql.org/docs/current/functions-aggregate.html#FUNCTIONS-HYPOTHETICAL-TABLE)
* Optimize now() query planning time
* Improve COPY insert performance
* Improve performance of UPDATE/DELETE on PG14 by excluding chunks

This release also includes several bug fixes.

If you are upgrading from a previous version and were using compression
with a non-default collation on a segmentby-column you should recompress
those hypertables.

**Features**
* #4045 Custom origin's support in CAGGs
* #4120 Add logging for retention policy
* #4158 Allow ANALYZE command on a data node directly
* #4169 Add support for chunk exclusion on DELETE to PG14
* #4209 Add support for chunk exclusion on UPDATE to PG14
* #4269 Continuous Aggregates finals form
* #4301 Add support for bulk inserts in COPY operator
* #4311 Support non-superuser move chunk operations
* #4330 Add GUC "bgw_launcher_poll_time"
* #4340 Enable now() usage in plan-time chunk exclusion

**Bug fixes**
* #3899 Fix segfault in Continuous Aggregates
* #4225 Fix TRUNCATE error as non-owner on hypertable
* #4236 Fix potential wrong order of results for compressed hypertable with a non-default collation
* #4249 Fix option "timescaledb.create_group_indexes"
* #4251 Fix INSERT into compressed chunks with dropped columns
* #4255 Fix option "timescaledb.create_group_indexes"
* #4259 Fix logic bug in extension update script
* #4269 Fix bad Continuous Aggregate view definition reported in #4233
* #4289 Support moving compressed chunks between data nodes
* #4300 Fix refresh window cap for cagg refresh policy
* #4315 Fix memory leak in scheduler
* #4323 Remove printouts from signal handlers
* #4342 Fix move chunk cleanup logic
* #4349 Fix crashes in functions using AlterTableInternal
* #4358 Fix crash and other issues in telemetry reporter

**Thanks**
* @abrownsword for reporting a bug in the telemetry reporter and testing the fix
* @jsoref for fixing various misspellings in code, comments and documentation
* @yalon for reporting an error with ALTER TABLE RENAME on distributed hypertables
* @zhuizhuhaomeng for reporting and fixing a memory leak in our scheduler
```
